### PR TITLE
fix(node): reset running when Start fails

### LINKED ISSUE #7678: StatusNode.running is not reset when Start fails, so a later Stop panics on a half-started node
`StatusNode.Start` marks the node running before startup can fail, and never rolls back. `Stop()` then assumes a fully-started node and dereferences fields that were never assigned.

**Set, never reset** — `pkg/backend/node/get_status_node.go:214`:

```go
if !n.running.CompareAndSwap(false, true) {
	n.logger.Debug("node is already running")
	return ErrNodeRunning
}
...
return n.startWithDB(config)   // no rollback of n.running on error
```

**Partial construction** — in `startWithDB`, `setupRPCClient()` runs at :352 and can return early; `n.downloader = ipfs.NewDownloader(...)` is only reached at :357. So a `setupRPCClient` failure leaves `running == true` with `downloader`, `rpcClient`, `timeSourceSrvc` and `accountsPublisher` all nil.

**The panic** — `Logout` gates on `IsRunning()` (`pkg/backend/geth_backend.go:2100`), sees `true`, and calls `Stop()`. `Stop()` dereferences without nil guards:

```go
n.timeSourceSrvc.Stop()      // :507
n.accountsPublisher.Close()  // :519
n.rpcClient.Stop()           // :521
n.downloader.Stop()          // :529
```

`n.mediaServer` and `n.localBackup` are nil-guarded; these four are not. On the failure path above, `timeSourceSrvc` panics first.

**Suggested fix.** Reset `n.running` to `false` when `startWithDB` returns an error, so `IsRunning()` reflects reality. Adding nil guards in `Stop()` treats the symptom — the underlying problem is that `running` is set before startup completes.

Severity: latent. It needs a start failure followed by a logout, and I have not traced a concrete production trigger — filing so it is recorded rather than because it is known to fire.

Found while reviewing #7676.

### PR DESCRIPTION
**Pull request by alexjba**
Fixes #7678

`StatusNode.Start` marks the node running via `CompareAndSwap` before `startWithDB` can fail, and never rolls back (`pkg/backend/node/get_status_node.go:214`).

`startWithDB` builds the node incrementally — a `setupRPCClient` failure at :352 returns before `n.downloader` is assigned at :357 — so a failed start leaves `running == true` with `rpcClient`, `timeSourceSrvc`, `accountsPublisher` and `downloader` all nil.

`Logout` gates teardown on `IsRunning()` (`pkg/backend/geth_backend.go:2100`), so it then calls `Stop()`, which dereferences those fields without nil guards and panics.

Rolling the flag back is the fix rather than nil-guarding `Stop()`: the defect is that `running` claims a startup that did not complete.

`TestStartFailureResetsRunning` fails on the parent commit.

### CODE REVIEW

File: pkg/backend/node/get_status_node.go [hunk 1]
```diff
@@ -222,7 +222,15 @@ func (n *StatusNode) Start(config *params.NodeConfig) error {
 
 	n.logger.Debug("starting with options", zap.Stringer("ClusterConfig", &config.ClusterConfig))
 
-	return n.startWithDB(config)
+	if err := n.startWithDB(config); err != nil {
+		// startWithDB builds the node incrementally, so a failure leaves it
+		// partially constructed. Roll the flag back: callers gate teardown on
+		// IsRunning, and Stop assumes every field was assigned.
+		n.running.Store(false)
```
Comment (saledjenic) [inline pkg/backend/node/get_status_node.go:236]: @alexjba this line now skips `statusNode.Stop()` entirely and everything `startWithDB` built before failing is never torn down. Unwind what was built, then `n.running.Store(false)`.

Comment (alexjba) [reply pkg/backend/node/get_status_node.go:236]: Done in b87e1dc9c

**Review by friofry — APPROVED** [review]
Left a couple of failing tests on your branch as #7719.

They cover an edge case: Start fails after initServices (not at TokenManager), then login retries on the same node. teardown leaves ethSrvc pointing at the RPC client it just stopped, so the second Start succeeds with a dead eth client.
